packages/core/contracts/oracle/implementation/SafetyModule.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.16;
+
+import "./Finder.sol";
+import "./GovernorV2.sol"; //todo: swap this to using an interface.
+import "./Constants.sol";
+import "../interfaces/StakerInterface.sol";
+import "./AdminIdentifierLib.sol";
+import "../../common/implementation/Lockable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract SafteyModule is Ownable, Lockable {
```

Comment:
now contract that lets someone propose an emergency action and ratify it. Note that this requires that the `emergencyActionAuthenticator` executes the action, thereby removing a lot of nasty things that could be done if this was a separate governance action, such as front running the deployment or the related attacks.